### PR TITLE
New structure to lookup element centroid 

### PR DESCRIPTION
**Pull request by aritorto**
This new structure LookUpCellCentroid is used in opm-simulators (eclbasevanguard) to search for the centroid of each element (cell) for any type of grid, including the CpGrid with/without LGRs.

### CODE REVIEW

**Comment by aritorto** [comment]
jenkins build this serial please

**Comment by aritorto** [comment]
jenkins build this serial please

**Comment by aritorto** [comment]
jenkins build this serial please

**Review by blattms — COMMENTED** [review]
Just a few more comments. Apart from the only two test files in CMakeLists_files.cmake but 3 added, these are minor.

File: CMakeLists_files.cmake [hunk 1]
```diff
@@ -99,6 +99,7 @@ list (APPEND TEST_SOURCE_FILES
   tests/cpgrid/zoltan_test.cpp
   tests/test_geom2d.cpp
   tests/test_gridutilities.cpp
+  tests/test_cellCentroid_polyhedralGrid.cpp
```
[the remark this answers is not in the document]
Comment (aritorto) [reply CMakeLists_files.cmake:102]: test_lookUpCellCentroid.cpp is old and only check CpGrid, so the idea was to delete it, and to have two separated tests: one for polyhedral grid and one for CpGrid.

File: opm/grid/LookUpCellCentroid.hh [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    return centroid;
+}
+
+template<typename Grid, typename GridView>
+template<typename GridType>
+typename std::enable_if_t<std::is_same_v<GridType,Dune::CpGrid>,std::array<double,3>>
+Opm::LookUpCellCentroid<Grid,GridView>::operator()(std::size_t elemIdx) const
+{
+    static_assert(std::is_same_v<Grid,GridType>);
+    const auto centroid =  this -> gridView_.grid().getEclCentroid(elemIdx); // Warning! might need to be changed, due to issue in simulators
```
[the remark this answers is not in the document]
Comment (aritorto) [reply opm/grid/LookUpCellCentroid.hh:128]: Yes, we'll do it.

File: opm/grid/cpgrid/CpGridData.cpp [hunk 3]
```diff
@@ -2262,44 +2262,24 @@ CpGridData::refinePatch(const std::array<int,3>& cells_per_dim, const std::array
...
-        }
-    }
-    return result;
-}
-
 std::array<double,3> CpGridData::computeEclCentroid(const int idx) const
 {
     const auto& cell_to_point_indices = this -> cell_to_point_[idx];
-    std::array<Dune::FieldVector<double, 3>,8> cell_to_point;
+    std::array<double,8> X;
```
[the remark this answers is not in the document]
Comment (aritorto) [reply opm/grid/cpgrid/CpGridData.cpp:2268]: Added!

File: tests/test_cellCentroid_polyhedralGrid.cpp [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    // Mapper
+    const Dune::MultipleCodimMultipleGeomTypeMapper<GridView> mapper(leafGridView, Dune::mcmgElementLayout());
+
+    for (const auto& element: Dune::elements(leafGridView)){
+        const int idx = mapper.index(element);
+        const auto& elemEclCentroid = eclGrid.getCellCenter(gridCartMapper.cartesianIndex(idx));
+        const auto& centroid = lookUpCellCentroid(idx);
+        for (int coord = 0; coord < 3; ++coord)
+        {
+            BOOST_CHECK_CLOSE(elemEclCentroid[coord], centroid[coord], 1e-9);
```
[the remark this answers is not in the document]
Comment (aritorto) [reply tests/test_cellCentroid_polyhedralGrid.cpp:106]: Changed!

File: CMakeLists_files.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (aritorto) [reply CMakeLists_files.cmake:102]: Now, the file test_lookUpCellCentroid.cpp is removed and does not appear in CMakeList_files.cmake

**Comment by aritorto** [comment]
jenkins build this serial please

**Pull request merged**